src/admin_ws.rs

```diff
@@ -0,0 +1,187 @@
+use std::{
+    future::Future,
+    sync::{Arc, Mutex},
+};
+
+use holochain_client::{AdminWebsocket, ConductorApiError, ConductorApiResult};
+use tokio::time::{sleep, Duration};
+
+use crate::{HcHttpGatewayError, HcHttpGatewayResult};
+
+const ADMIN_WS_CONNECTION_MAX_RETRIES: usize = 1;
+const ADMIN_WS_CONNECTION_RETRY_DELAY_MS: u64 = 1000;
+
+/// A wrapper around AdminWebsocket that automatically handles reconnection
+/// when the connection is lost due to network issues or other failures.
+#[derive(Clone)]
+pub struct ReconnectingAdminWebsocket {
+    /// The WebSocket URL to connect to
+    url: String,
+    /// The handle to the AdminWebsocket connection
+    connection_handle: Arc<Mutex<Option<AdminWebsocket>>>,
+    /// Current retry attempt counter
+    current_retries: usize,
+}
+
+impl ReconnectingAdminWebsocket {
+    /// Creates a new ReconnectingAdminWebsocket with the specified parameters
+    pub fn new(url: &str) -> Self {
+        ReconnectingAdminWebsocket {
+            url: url.to_string(),
+            connection_handle: Arc::new(Mutex::new(None)),
+            current_retries: 0,
+        }
+    }
+
+    /// Checks if there is an active connection
+    fn is_connected(&self) -> HcHttpGatewayResult<bool> {
+        let connection = self.connection_handle.lock().map_err(|e| {
+            HcHttpGatewayError::InternalError(format!(
+                "Mutex was poisoned during is_connected check: {}",
```

Comment:
This only happens if there has been a panic elsewhere while the lock was held and it's not recoverable. It's letting you handle the error in case you want to replace the mutex or otherwise resolve the problem state yourself but bubbling the state up is not solving the issue. I'd recommend using `.expect("some message")` here 

https://doc.rust-lang.org/std/sync/struct.Mutex.html#poisoning